extensions/servlet/test/com/google/inject/servlet/ServletDefinitionPathsTest.java

```diff
@@ -216,35 +229,39 @@ protected void service(HttpServletRequest servletRequest,
 
   // Data-driven test.
   public final void testPathInfoWithRegexMatching() throws IOException, ServletException {
+
     // first a mapping of /*
-    pathInfoWithRegexMatching("/path/index.html", "/path", "/(.)*", "/index.html", "");
-    pathInfoWithRegexMatching("/path//hulaboo///index.html", "/path", "/(.)*",
+    pathInfoWithRegexMatching("/path/index.html", "/path", "(/).*", "/index.html", "");
```

Comment:
I think it goes like this:

the regex pattern matcher returns the remaining path "behind - 1" the last group.
before my fix servletPath = "" - matched against /(.)* did return nothing because no match was found.

after my fix servlerPath is set to the reuqest URI without the context path. serlvetPath = "/index.html" -> matches against /(.)* and returns "l".

This is why I'm pretty sure the given regular expression in the unit tests are wrong, especially for all test cases with servletPath = "".

What do you think? I'm I right or do I have a thinko here?